Code/GraphMol/FindStereo.cpp

```diff
@@ -0,0 +1,445 @@
+        // set "fake stereo"
+        ochiralTypes[atom->getIdx()] = atom->getChiralTag();
+        atom->setChiralTag(Atom::CHI_TETRAHEDRAL_CW);
+        atomSymbols[aidx] =
+            (boost::format("%s-%d") % atom->getSymbol() % atom->getIdx()).str();
+      } else {
+        atomSymbols[aidx] = atom->getSymbol();
+      }
+    } else {
+      atomSymbols[aidx] = atom->getSymbol();
+    }
+  }
+
+  std::vector<std::string> bondSymbols(mol.getNumBonds());
+  boost::dynamic_bitset<> knownBonds(mol.getNumBonds());
+  boost::dynamic_bitset<> possibleBonds(mol.getNumBonds());
+  for (const auto bond : mol.bonds()) {
+    auto bidx = bond->getIdx();
+    if (detail::isBondPotentialStereoBond(bond)) {
+      auto sinfo = detail::getStereoInfo(bond);
+      switch (sinfo.specified) {
+        case Chirality::StereoSpecified::Unknown:
+        case Chirality::StereoSpecified::Specified:
+          knownBonds.set(bidx);
+          break;
+        case Chirality::StereoSpecified::Unspecified:
+          break;
+        default:
+          throw ValueErrorException("bad StereoInfo.specified type");
+      }
+      if (flagPossible ||
+          sinfo.specified != Chirality::StereoSpecified::Unspecified) {
+        possibleBonds.set(bidx);
+        bondSymbols[bidx] =
+            (boost::format("%s-%d") % getBondSymbol(bond) % bond->getIdx())
+                .str();
+      } else {
+        bondSymbols[bidx] = getBondSymbol(bond);
+      }
+    } else {
+      bondSymbols[bidx] = getBondSymbol(bond);
+    }
+  }
+
+  // we will use the canonicalization code, pretending that each potential
+  // stereo atom and bond is specified and different from all others. After
+  // we've done that we can re-examine the potential stereo atoms and bonds and
+  // remove any where two controlling atoms have the same rank
+  boost::dynamic_bitset<> atomsInPlay(mol.getNumAtoms());
+  atomsInPlay.set();
+  boost::dynamic_bitset<> bondsInPlay(mol.getNumBonds());
+  bondsInPlay.set();
+  std::vector<unsigned int> aranks;
+  bool breakTies = false;
+  bool includeChirality = true;
+  bool includeIsotopes = false;
+  Canon::rankFragmentAtoms(mol, aranks, atomsInPlay, bondsInPlay, &atomSymbols,
+                           &bondSymbols, breakTies, includeChirality,
+                           includeIsotopes);
+  std::vector<StereoInfo> res;
+  for (const auto atom : mol.atoms()) {
+    auto aidx = atom->getIdx();
+    if (ochiralTypes.find(aidx) != ochiralTypes.end()) {
+      atom->setChiralTag(ochiralTypes[aidx]);
+    }
+    if (possibleAtoms[aidx]) {
+      auto sinfo = detail::getStereoInfo(atom);
+      std::vector<unsigned int> nbrs;
+      nbrs.reserve(sinfo.controllingAtoms.size());
+      bool haveADupe = false;
+      for (auto nbrIdx : sinfo.controllingAtoms) {
+        auto rnk = aranks[nbrIdx];
+        if (std::find(nbrs.begin(), nbrs.end(), rnk) != nbrs.end()) {
+          haveADupe = true;
+          break;
+        } else {
+          nbrs.push_back(rnk);
```

Comment:
We don't need to, like, redo the ranking if we found an achiral atom, do we?

Reply:
oh, excellent catch.
This is incorrect:
```
In [2]: ps = Chem.FindPotentialStereo(Chem.MolFromSmiles('FC(Cl)(F)C(C(Cl)(F)F)I'))                          

In [3]: len(ps)                                                                                              
Out[3]: 1
```